libs/platform/src/lib/components/thumbnail/thumbnail-details/thumbnail-details.component.ts

```diff
@@ -0,0 +1,145 @@
+
+    /** @hidden Start index of currently active items */
+    currentActiveSlidesStartIndex = 0;
+
+    /** @hidden Make left navigation button disabled */
+    leftButtonDisabled = false;
+
+    /** @hidden Make right navigation button disabled */
+    rightButtonDisabled = false;
+
+    /** @hidden */
+    constructor(private readonly _elementRef: ElementRef, public dialogRef: DialogRef,
+        private _cdr: ChangeDetectorRef
+    ) {}
+
+    /** @hidden */
+    ngOnInit(): void {
+        this.dialogRef.data.selectedMedia.selected = true;
+        this.currentActiveSlidesStartIndex = this.dialogRef.data.mediaList.indexOf(this.dialogRef.data.selectedMedia);
+    }
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        this.currentActiveSlidesStartIndex = this.dialogRef.data.mediaList.indexOf(this.dialogRef.data.selectedMedia);
+        if (this.dialogRef.data.mediaList.length > 0) {
+            this._buttonVisibility();
+        } else {
+            this.leftButtonDisabled = true;
+            this.rightButtonDisabled = true;
+            this._cdr.detectChanges();
+        }
+    }
+
+    /** Transitions to the previous image in the thumbnail. */
+    previous(): void {
+        if (this.currentActiveSlidesStartIndex <= 0) {
+            return;
+        } else {
+            this.currentActiveSlidesStartIndex = this.currentActiveSlidesStartIndex - 1;
+            this.dialogRef.data.selectedMedia = this.dialogRef.data.mediaList[this.currentActiveSlidesStartIndex];
+            this.dialogRef.data.mediaList.forEach(item => item.selected = false);
+            this.dialogRef.data.mediaList[this.currentActiveSlidesStartIndex].selected = true;
+        }
+        this._buttonVisibility();
+    }
+
+    /** Transitions to the next image in the thumbnail. */
+    next(): void {
+        if (this.currentActiveSlidesStartIndex >= this.dialogRef.data.mediaList.length - 1) {
+            return;
+        } else {
+            this.currentActiveSlidesStartIndex = this.currentActiveSlidesStartIndex + 1;
+            this.dialogRef.data.selectedMedia = this.dialogRef.data.mediaList[this.currentActiveSlidesStartIndex];
+            this.dialogRef.data.mediaList.forEach(item => item.selected = false);
+            this.dialogRef.data.mediaList[this.currentActiveSlidesStartIndex].selected = true;
+        }
+        this._buttonVisibility();
+
+    }
+
+    /** handles click on thumbnail images  */
+    thumbnailClickHandle(selectedMedia: Media): void {
+        this.dialogRef.data.selectedMedia = selectedMedia;
+        this.currentActiveSlidesStartIndex = this.dialogRef.data.mediaList.indexOf(this.dialogRef.data.selectedMedia);
+        this._buttonVisibility();
+    }
+
+    /**Close dialog */
+    closeDialog(): void {
+        if (this.dialogRef.data.mediaList.length > this.dialogRef.data.maxImages) {
+            this.dialogRef.data.mediaList[this.dialogRef.data.maxImages - 1].overlayRequired = true;
+        }
```

Comment:
```suggestion
        const { mediaList } = this.dialogRef.data;
        const { maxImages } = this.dialogRef.data;
        if (mediaList.length > maxImages) {
            mediaList[maxImages - 1].overlayRequired = true;
        }
```

Reply:
These value can't be declared as constant, as dialogRef only available once the dialog opens if we declare it as constant it will throw compilation error as dialogRef is undefined.